src/main/java/duke/tool/Parser.java

```diff
@@ -0,0 +1,110 @@
+package duke.tool;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.exceptions.NoDescriptionException;
+import duke.exceptions.NoSuchOrderException;
+import duke.exceptions.NoTaskChosenException;
+import duke.exceptions.NoThisNumOfTaskException;
+import duke.exceptions.NoTimeException;
+import duke.tasks.Deadline;
+import duke.tasks.Delete;
+import duke.tasks.Done;
+import duke.tasks.Event;
+import duke.tasks.Exit;
+import duke.tasks.Find;
+import duke.tasks.List;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * The class deals with making sense of the user command.
+ */
+public class Parser {
+    public static DateTimeFormatter VALID_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    /**
+     * Check whether the characters in the string represents an integer.
+     * @param s
+     * @return
+     */
+    public static boolean isInteger(String s) {
+        if (s == null) {
+            return false;
+        }
+        try {
+            int d = Integer.parseInt(s);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Parse the command from user and return the corresponding task.
+     * @param order the order from user.
+     * @param tl
+     * @return the task which can be understood by Duke.
+     */
+    public static Task parse(String order, TaskList tl) {
+        try {
+            int numOfOrders = tl.getNumOfTasks();
+            if (order != null) {
+                String[] command = order.split(" ", 2);
+                if (ValidInput.getCmdType(command[0]).name().equals("EXIT")) {
```

Comment:
Perhaps you can also consider bringing this value (ValidInput.getCmdType(command[0]).name()) out of the loop and assigning it as a String.  For example, 
String s = ValidInput.getCmdType(command[0]).name(). 
If s.equals("EXIT") {
    //do something
}